devito/tools/decorators.py

```diff
@@ -203,3 +203,10 @@ def __enter__(self):
     def __exit__(self, *args):
         self.timings[self.name] = time() - self.tic
         del timed_pass.timings[get_ident()]
+        try:
```

Comment:
again, I had a detailed explanation and the browser crashed, sob...

Basically: we don't need to wrap the Operator construction with a try-except because there's no global state (!) and therefore any exceptions arising during compilation can be propagated to the user without the need for any cleanup code...

... aside from this profiling-related stuff. But that's exactly why this class implements a context manager. This `__exit__` method _always_ gets executed (even in presence of exceptions), so we can execute the cleanup here safely, without exposing the internals of this class to the outside world (the Operator class, in our case...)